src/structures/gdt.rs

```diff
@@ -306,17 +307,62 @@ impl Descriptor {
     /// Creates a TSS system descriptor for the given TSS.
     #[inline]
     pub fn tss_segment(tss: &'static TaskStateSegment) -> Descriptor {
+        // SAFETY: if iomap_size is zero, there are no requirements to uphold.
+        unsafe { Self::tss_segment_raw(tss, 0) }
+    }
+
+    /// Creates a TSS system descriptor for the given TSS, setting up the IO permissions bitmap.
+    pub fn tss_segment_with_iomap(
+        tss: &'static TaskStateSegment,
+        iomap: &'static [u8],
+    ) -> Result<Descriptor, InvalidIoMap> {
+        if iomap.len() > 8193 {
+            return Err(InvalidIoMap::TooLong { len: iomap.len() });
+        }
+
+        let base = iomap.as_ptr() as usize - tss as *const _ as usize;
```

Comment:
We need to use wrapping_offset_from here and get an isize. We need to deal with isizes because the iomap could come before the tss, and that would also be wrong.

Reply:
`wrapping_offset_from` doesn't exist. I think that `offset_from` would also be incorrect, as according to `offset_from`'s safety contract:

```
The distance between the pointers, in bytes, cannot overflow an isize.
```

This is overflowed if TSS is higher half and IO permissions bitmap is lower half, I think.

Reply:
I've just handled this with a manual `if` check.